tmail-backend/mailbox/opensearch/src/main/java/com/linagora/tmail/mailbox/opensearch/TmailOpenSearchMailboxConfiguration.java

```diff
@@ -0,0 +1,111 @@
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ *                                                                  *
+ *  This file was taken and adapted from the Apache James project.  *
+ *                                                                  *
+ *  https://james.apache.org                                        *
+ *                                                                  *
+ *  It was originally licensed under the Apache V2 license.         *
+ *                                                                  *
+ *  http://www.apache.org/licenses/LICENSE-2.0                      *
+ ********************************************************************/
+
+package com.linagora.tmail.mailbox.opensearch;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.commons.configuration2.Configuration;
+
+public class TmailOpenSearchMailboxConfiguration {
+    public static class Builder {
+        private Optional<Boolean> subjectNgramEnabled;
+        private Optional<Boolean> subjectNgramHeuristicEnabled;
+
+        Builder() {
+            subjectNgramEnabled = Optional.empty();
+            subjectNgramHeuristicEnabled = Optional.empty();
+        }
+
+        public Builder subjectNgramEnabled(Boolean subjectNgramEnabled) {
+            this.subjectNgramEnabled = Optional.ofNullable(subjectNgramEnabled);
+            return this;
+        }
+
+        public Builder subjectNgramHeuristicEnabled(Boolean subjectNgramHeuristicEnabled) {
+            this.subjectNgramHeuristicEnabled = Optional.ofNullable(subjectNgramHeuristicEnabled);
+            return this;
+        }
+
+        public TmailOpenSearchMailboxConfiguration build() {
+            return new TmailOpenSearchMailboxConfiguration(
+                subjectNgramEnabled.orElse(DEFAULT_SUBJECT_NGRAM_ENABLED),
+                subjectNgramHeuristicEnabled.orElse(DEFAULT_SUBJECT_NGRAM_HEURISTIC_ENABLED)
+            );
+        }
+    }
+
+    public static Builder builder() {
+        return new TmailOpenSearchMailboxConfiguration.Builder();
+    }
+
+    public static TmailOpenSearchMailboxConfiguration fromProperties(Configuration configuration) {
+        return builder()
+            .subjectNgramEnabled(configuration.getBoolean(SUBJECT_NGRAM_ENABLED, null))
+            .subjectNgramHeuristicEnabled(configuration.getBoolean(SUBJECT_NGRAM_HEURISTIC_ENABLED, null))
+            .build();
+    }
+
+    private static final String SUBJECT_NGRAM_ENABLED = "subject.ngram.enabled";
+    private static final String SUBJECT_NGRAM_HEURISTIC_ENABLED = "subject.ngram.heuristic.enabled";
+    private static final boolean DEFAULT_SUBJECT_NGRAM_ENABLED = false;
+    private static final boolean DEFAULT_SUBJECT_NGRAM_HEURISTIC_ENABLED = false;
```

Comment:
```suggestion
    private static final boolean DEFAULT_SUBJECT_NGRAM_DISABLED = false;
    private static final boolean DEFAULT_SUBJECT_NGRAM_HEURISTIC_DISABLED = false;
```

Reply:
Not really I think... disabled = false could mean maybe enabled for some people? Two negatives = positive? Well I interpret it that way :)